src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets

```diff
@@ -238,6 +238,10 @@ Copyright (c) .NET Foundation. All rights reserved.
       <PackageReference Include="Microsoft.Net.Compilers.Toolset.Framework" PrivateAssets="all" Version="$(_NetFrameworkHostedCompilersVersion)" IsImplicitlyDefined="true" />
     </ItemGroup>
 
+    <Target Name="_AddGenerateSbomTask" BeforeTargets="CollectPackageReferences">
+      <ItemGroup Condition=" '$(GenerateSbom)' == 'true' ">
+        <PackageReference Include="Microsoft.Sbom.Targets" Version="0.0.0-preview.0" IsImplicitlyDefined="true" />
+      </ItemGroup>
```

Comment:
Ah - you put this _inside_ another target. Move this entire chunk down after line 246 and add a closing `</Target>`